aten/src/ATen/TensorImpl.cpp

```diff
@@ -62,13 +95,19 @@ TensorImpl::TensorImpl(Storage&& storage, TensorTypeId type_id, ScalarType scala
       numel_(0),
       type_id_(type_id),
       scalar_type_(scalar_type),
-      is_variable_(is_variable) {}
+      is_variable_(is_variable),
+      has_storage_(has_storage),
+      has_strides_(has_strides),
+      support_resize_by_maybe_zero_dim_(support_resize_by_maybe_zero_dim) {}
 
 IntList TensorImpl::sizes() const {
   return sizes_;
 }
 
 IntList TensorImpl::strides() const {
+  if (!has_strides_) {
+    AT_ERROR("This type of tensor does not have strides");
```

Comment:
It would be nice to also say what the type of the tensor is.

Reply:
What's the best way to get the tensor type? (Is using TensorTypeId still a good way to go?)